src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -56,9 +56,14 @@
 _COMMON_TENANT = 'common'
 
 
-def _authentication_context_factory(authority, cache):
+def _authentication_context_factory(tenant, cache):
     import adal
-    return adal.AuthenticationContext(authority, cache=cache, api_version=None)
+    authority_url = CLOUD.endpoints.active_directory
+    is_adfs = authority_url.lower().endswith('/adfs')
+    if not is_adfs:
+        authority_url = authority_url + '/' + (tenant or _COMMON_TENANT)
```

Comment:
In case of ADFS, we should be ignoring --tenant parameter, even if user passes some value. I believe we are already doing that here, pls confirm